Flow.Launcher.Core/Plugin/JsonRPCPluginSettings.cs

```diff
@@ -23,26 +19,44 @@ public class JsonRPCPluginSettings
 
-        // maybe move to resource?
-        private static readonly Thickness settingControlMargin = new(0, 9, 18, 9);
-        private static readonly Thickness settingCheckboxMargin = new(0, 9, 9, 9);
-        private static readonly Thickness settingPanelMargin = new(0, 0, 0, 0);
-        private static readonly Thickness settingTextBlockMargin = new(70, 9, 18, 9);
-        private static readonly Thickness settingLabelPanelMargin = new(70, 9, 18, 9);
-        private static readonly Thickness settingLabelMargin = new(0, 0, 0, 0);
-        private static readonly Thickness settingDescMargin = new(0, 2, 0, 0);
-        private static readonly Thickness settingSepMargin = new(0, 0, 0, 2);
+        private static readonly Thickness SettingPanelMargin = (Thickness)Application.Current.FindResource("SettingPanelMargin");
+        private static readonly Thickness SettingPanelItemLeftMargin = (Thickness)Application.Current.FindResource("SettingPanelItemLeftMargin");
+        private static readonly Thickness SettingPanelItemTopBottomMargin = (Thickness)Application.Current.FindResource("SettingPanelItemTopBottomMargin");
+        private static readonly Thickness SettingPanelItemLeftTopBottomMargin = (Thickness)Application.Current.FindResource("SettingPanelItemLeftTopBottomMargin");
+        private static readonly double SettingPanelTextBoxMinWidth = (double)Application.Current.FindResource("SettingPanelTextBoxMinWidth");
+        private static readonly double SettingPanelPathTextBoxWidth = (double)Application.Current.FindResource("SettingPanelPathTextBoxWidth");
+        private static readonly double SettingPanelAreaTextBoxMinHeight = (double)Application.Current.FindResource("SettingPanelAreaTextBoxMinHeight");
 
         public async Task InitializeAsync()
         {
-            _storage = new JsonStorage<ConcurrentDictionary<string, object>>(SettingPath);
-            Settings = await _storage.LoadAsync();
+            if (Settings == null)
+            {
+                _storage = new JsonStorage<ConcurrentDictionary<string, object?>>(SettingPath);
+                Settings = await _storage.LoadAsync();
+
+                // Because value type of settings dictionary is object which causes them to be JsonElement when loading from json files,
+                // we need to convert it to the correct type
+                foreach (var (key, value) in Settings)
+                {
+                    if (value is JsonElement jsonElement)
+                    {
+                        Settings[key] = jsonElement.ValueKind switch
+                        {
+                            JsonValueKind.String => jsonElement.GetString() ?? value,
+                            JsonValueKind.True => jsonElement.GetBoolean(),
+                            JsonValueKind.False => jsonElement.GetBoolean(),
+                            JsonValueKind.Null => null,
+                            _ => value
+                        };
+                    }
```

Comment:
I would really like to avoid as much nesting as possible, so could you please invert the conditional?

```suggestion
                    if (value is not JsonElement jsonElement) continue;

                    Settings[key] = jsonElement.ValueKind switch
                    {
                        JsonValueKind.String => jsonElement.GetString() ?? value,
                        JsonValueKind.True => jsonElement.GetBoolean(),
                        JsonValueKind.False => jsonElement.GetBoolean(),
                        JsonValueKind.Null => null,
                        _ => value
                    };
```